[keaganpzh] iP

keaganpzh opened the pull request:
### Rat

> all my friends are rats.

Rat is your personal task manager built on the ~~latest~~ version of everyone's _favourite_ programming language, Java 11!

### Features

- [x] Greets you
- [x] Records your tasks
- [x] ???

### Installation

1. Download the latest version of Rat [here](https://github.com/keaganpzh/ip/releases/A-BetterGui/Rat-1.1.jar). 
2. Open a new Terminal in the location folder of the download. Run the following:
```shell
java -jar Rat-1.1.jar
``` 
3. Type 'help' for a list of commands.
4. Profit 🤩

### For Java Nerds

Here's what the `main` method of Rat looks like:
```java
public static void main(String[] args) {
    Rat.initialise();
    ratInput.handleInput();
    sc.close();
}
``` 

### New in v1.1

- Improved GUI for Rat
- Note-taking functionality

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

keaganpzh committed:
.gitignore: include .class files

keaganpzh committed:
feat: Renamed bot, implemented greeting and exit message.

keaganpzh committed:
fix: edit test files to match new class

keaganpzh committed:
refactor: created new class to handle console printing

keaganpzh committed:
feat: echo feature using Java.util.Scanner

keaganpzh committed:
feat: Level-2: created classes for storage and task and handle inputs accordingly

keaganpzh committed:
feat: Level-3 -> able to mark tasks; new class to handle scanner inputs

keaganpzh committed:
feat: Level-4 -> created task subclasses and handled inputs

keaganpzh committed:
test: A-TextUiTesting -> Created input.txt and expected

keaganpzh committed:
fix: github actions

keaganpzh committed:
fix: github actions commands

keaganpzh committed:
fix: github actions

keaganpzh committed:
fix: github actions again

keaganpzh committed:
fix: github actions (fr this time)

keaganpzh committed:
A-TextUiTesting -> Setup testing and workflow

keaganpzh committed:
feat: Level-5 -> Error handling when handling inputs implemented.

keaganpzh committed:
feat: Level-6 -> Implemented Deletions for Tasks.

keaganpzh committed:
feat: added "delete all" and "help" command

keaganpzh committed:
feat: added "delete all" and "help" commands and functionality

keaganpzh committed:
fix: removed color for text for test compatibility

keaganpzh committed:
fix: Input handler handles system exit instead of throwing exception

keaganpzh committed:
docs: add javadocs

keaganpzh committed:
refactor: used static imports for UI printer methods

keaganpzh committed:
refactor: package names

keaganpzh committed:
fix: update test files to reflect change in package structure

keaganpzh committed:
feat: able to format data and write to storage

keaganpzh committed:
feat: Level-7 -> Able to read from file on startup

keaganpzh committed:
Merge branch 'Level-7' to master

keaganpzh committed:
feat: Level-8 -> Dates are now stored as java.time.LocalDateTime objects

keaganpzh committed:
Merge branch 'branch-Level-8' into master

keaganpzh committed:
A-MoreOOP: separate commands into classes

keaganpzh committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

keaganpzh committed:
A-Gradle: Setup gradle

keaganpzh committed:
Merge branch 'branch-A-Gradle' into master

keaganpzh committed:
A-JUnit: create testcases for Rat. automate with gradle

keaganpzh committed:
A-Jar: build Rat.jar with gradle

keaganpzh committed:
Level-9: add find functionality

keaganpzh committed:
Level-9: add JavaDoc for FindCommand.java

keaganpzh committed:
A-JavaDoc: add JavaDoc for all files under src/main/java/rat/

keaganpzh committed:
A-CodingStandard: follow CS2103T coding standards

keaganpzh committed:
Merge branch 'branch-A-JavaDoc' into master

keaganpzh committed:
Merge branch 'branch-A-CodingStandard' into master

keaganpzh committed:
A-CheckStyle: setup checkstyle for gradle and fixed style violations

keaganpzh committed:
Merge branch 'branch-A-CheckStyle' into master

keaganpzh committed:
Refactor: refactored commands in preparation for GUI implementation

keaganpzh committed:
Level-10: Create JavaFX GUI for Rat

keaganpzh committed:
Merge branch 'branch-Level-10' into master

keaganpzh committed:
refactor JavaDocs and DialogBox

keaganpzh committed:
test commit for actions

keaganpzh committed:
Actions: change java distro

keaganpzh committed:
Actions: update Java distro for JavaFX support

keaganpzh committed:
A-Varargs: update methods to support varargs

Kb-Tay commented on src/main/java/rat/tasks/Task.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * The status of the task.
+     */
+    private boolean isDone;
```
Appropiate naming of boolean field

Kb-Tay replied:
Good job on following standards for boolean fields

Kb-Tay commented on src/test/java/rat/command/RatCommandTest.java:
```diff
@@ -0,0 +1,44 @@
+     * Tests the validateTime method with a valid input.
+     */
+    @Test
+    public void validateTime_validTime_success() {
+        try {
+            assertTrue(RatCommand.validateTime("01/01/2020 00:00"));
+        } catch (ParseException e) {
+            fail();
+        }
+    }
+
+    /**
+     * Tests the validateTime method with an invalid input.
+     */
+    @Test
+    public void validateTime_invalidTime_exceptionThrown() {
+        try {
+            assertFalse(RatCommand.validateTime("01 Jan 2023"));
+        } catch (ParseException e) {
+            assertThrows(ParseException.class, () -> {
+                throw e;
+            });
+        }
+    }
```
Good job on following standards for test functions testing for exceptions.

Kb-Tay commented on src/main/java/rat/Rat.java:
```diff
@@ -0,0 +1,83 @@
+     * @param input The user's input.
+     * @return The response to the user's input generated from respective commands.
+     */
+    public String getResponse(String input) {
+        return ratInput.handleInput(input);
+    }
+
+    public String getWelcome() {
+        return "Hello! I'm Rat! What can I do for you? Type \"help\" to see a list of commands";
+    }
```
Good job on using verbs to name functions.

Kb-Tay commented on src/main/java/rat/tasks/Event.java:
```diff
@@ -0,0 +1,80 @@
+    /**
+     * The start time of the task, represented by a LocalDateTime object.
+     */
+    private LocalDateTime startTime;
+
+    /**
+     * The end time of the task, represented by a String.
+     */
+    private LocalDateTime endTime;
+
+    /**
+     * The start time of the task, represented by a String.
+     * The date and time is formatted as "EEE, d MMM yyyy HH:mm".
+     */
+    private String startTimeString;
+
+    /**
+     * The end time of the task, represented by a String.
+     * The date and time is formatted as "EEE, d MMM yyyy HH:mm".
+     */
+    private String endTimeString;
```
Good job using camel casing for naming of fields.

Kb-Tay commented on src/main/java/rat/command/RatCommand.java:
```diff
@@ -0,0 +1,51 @@
+        this.ratTaskManager = ratTaskManager;
+    }
+
+    /**
+     * Executes the command.
+     */
```
What does executing the command do? Should the java docs give more details on what the execute function does?

keaganpzh committed:
fix: update build.gradle to support jfx packaging

ceilingFan456 commented on src/main/java/rat/tasks/Deadline.java:
```diff
@@ -10,16 +13,25 @@ public class Deadline extends Task {
     protected Deadline(String deadline, String name) {
+        super(name);
+        this.deadline = LocalDateTime.parse(deadline, DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));
+        this.deadlineString = this.deadline.format(DateTimeFormatter.ofPattern("EEE, d MMM yyyy HH:mm"));
```
Good job on following java naming convention and making the datetime processing really concise.

ceilingFan456 commented on src/main/java/rat/io/RatInput.java:
```diff
@@ -165,11 +175,15 @@ protected void handleEvent(String params) {
             printWithLines("Event name cannot be empty");
         } catch (ArrayIndexOutOfBoundsException e) {
             printWithLines("Invalid event format. Please use \"event <name> /from <start> /to <end>\"");
+        } catch (ParseException e) {
```
I like how you handle all the exceptions together.

ceilingFan456 commented on src/main/java/rat/tasks/Event.java:
```diff
@@ -37,13 +54,13 @@ protected Event(String startTime, String endTime, String name) {
     public String toString() {
         String taskType = "[E]";
-        return taskType + super.toString() + " (from: " + this.startTime + " to: " + this.endTime + ")";
+        return taskType + super.toString() + " (from: " + this.startTimeString + " to: " + this.endTimeString + ")";
```
Good job on changing the name to make the type more obvious.

ceilingFan456 commented on src/main/java/rat/tasks/RatTaskManager.java:
```diff
@@ -47,10 +48,10 @@ protected void getDataFromFile() {
                 this.taskList.add(new ToDo(taskData[2]));
```
Will having a taskData class instead of hardcoding the indices to handle the conversion better for future format changes? It might not be necessary but a good improvement to have.

ceilingFan456 reviewed (commented):
A really good job of implementing the DateTime format concisely and following the Java coding conventions.

AnnabelTing commented on src/main/java/rat/Rat.java:
```diff
@@ -0,0 +1,83 @@
+     * @param input The user's input.
+     * @return The response to the user's input generated from respective commands.
+     */
+    public String getResponse(String input) {
```
Good job on naming the getters!

AnnabelTing commented on src/main/java/rat/command/AddCommand.java:
```diff
@@ -0,0 +1,136 @@
+
+    @Override
+    public String getResponse() {
+        switch (commandType) {
```
Great job on following the coding style!

AnnabelTing commented on src/test/java/rat/tasks/ToDoTest.java:
```diff
@@ -0,0 +1,52 @@
+     * Tests the creation of a ToDo task.
+     */
+    @Test
+    public void testToDo_create() {
```
Perhaps you can consider including the expected outcome of the test?

AnnabelTing reviewed (commented):
Overall LGTM

keaganpzh committed:
Add Java Assertions

There are no assertions implemented at this point in time. There may be
cases where methods are called on null objects.

Implement assertions at rat.gui.MainWindow and rat.tasks.RatTaskManager.

Assertions are a good way to document important assumptions that should
hold true at certain points of the code.

keaganpzh committed:
Merge pull request #1 from keaganpzh/branch-A-Assertions

PR to merge branch-A-Assertions to master

keaganpzh committed:
Refactor to improve code quality

Several violations of code quality were found throughout the code, which
impacts readability and safety.

Remove 'magic numbers', introduce new variables to avoid reassigned
parameters

To follow the coding standards provided by the course.

keaganpzh committed:
A-CI: setup Github Actions to run JUnit and Checkstyle

keaganpzh committed:
Merge pull request #2 from keaganpzh/branch-A-CodeQuality

PR to merge branch-A-CodeQuality into master

keaganpzh committed:
refactor: fix checkstyle

keaganpzh committed:
Merge pull request #3 from keaganpzh/branch-A-CI

PR to merge branch-A-CI intoto master

keaganpzh committed:
Feature: D-Notes

Implemented a new feature of the extension D-Notes. Rat can now track
user notes.

Create new class to handle creation and deletion of notes. Implement new
testcases to test note functions.

keaganpzh committed:
Merge pull request #4 from keaganpzh/branch-D-Notes

PR to merge branch-D-Notes into master

keaganpzh committed:
A-MoreErrorHandling: validate time for events such that end time > start time

keaganpzh committed:
A-BetterGui: Round user and bot images

keaganpzh committed:
Merge branch 'branch-A-BetterGui' into master

keaganpzh committed:
fix: typo in response for adding task

keaganpzh committed:
fix: squash bugs for saving tasks and notes to file

keaganpzh committed:
docs: initial commit

keaganpzh committed:
fix: squash bugs for find command

keaganpzh committed:
update documentation

keaganpzh committed:
update documentation: fix tables attempt

keaganpzh committed:
migrate deployment of docs to separate branch

keaganpzh committed:
update docs

keaganpzh committed:
fix docs: picture not showing

keaganpzh committed:
Merge branch 'branch-docs' into master

keaganpzh committed:
docs: fix table

keaganpzh committed:
try fix table again

keaganpzh committed:
update repo README

keaganpzh committed:
Merge branch 'branch-docs' into master
